.github/workflows/lint.yml

```
@@ -0,0 +1,68 @@
+name: AMR-Wind Lint
+
+on:
+  push:
+    branches: [development]
+    paths:
+      - 'cmake/**'
+      - 'amr-wind/**'
+      - 'submods/**'
+      - 'test/**'
+      - 'unit_tests/**'
+      - 'CMakeLists.txt'
+      - '.github/workflows/lint.yml'
+  pull_request:
+    branches: [development]
+    paths:
+      - 'cmake/**'
+      - 'amr-wind/**'
+      - 'submods/**'
+      - 'test/**'
+      - 'unit_tests/**'
+      - 'CMakeLists.txt'
+      - '.github/workflows/lint.yml'
+
+jobs:
+  Lint:
+    runs-on: macos-latest
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{github.token}}
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+      - name: Dependencies
+        run: brew install cppcheck
+      - name: Configure
+        run: |
+          echo "NPROCS=$(sysctl -n hw.ncpu)" >> $GITHUB_ENV
+          cmake \
+          -Bbuild-lint \
+          -DAMR_WIND_ENABLE_MPI:BOOL=OFF \
+          -DAMR_WIND_ENABLE_TESTS:BOOL=ON \
+          -DAMR_WIND_TEST_WITH_FCOMPARE:BOOL=OFF \
+          -DAMR_WIND_ENABLE_MASA:BOOL=OFF \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON \
+          ${{github.workspace}}
+      - name: Check
+        working-directory: build-lint
+        run: |
+          # Sanitize AMREX_GIT_VERSION define with 3 backslashes and quotes which cppcheck can't parse
+          sed -i.bak 's/\\\\\\"//g' compile_commands.json
```

Why do we need a backup file here? Is this to workaround BSD sed's `-i` option?
```suggestion
          sed -i'' 's/\\\\\\"//g' compile_commands.json
```

---

I've never been able to do in place on on MacOS.
```
% sed -i'' 's/\\\\\\"//g' compile_commands.json
sed: 1: "compile_commands.json": command c expects \ followed by text
```

---

```
❯ uname -s
Darwin
❯ echo '-DAMREX_GIT_VERSION=\\\\\\"20.12-9-g5fb87ea35fa5\\\\\\"' > test.txt
❯ cat test.txt
-DAMREX_GIT_VERSION=\\\"20.12-9-g5fb87ea35fa5\\\"
❯ sed -i '' -e 's/\\\\\\"//g' test.txt
❯ cat test.txt
-DAMREX_GIT_VERSION=20.12-9-g5fb87ea35fa5
```

---

You forgot `-e` before the replacement string

```
sed -i'' 's/\\\\\\"//g' compile_commands.json
```

---

Do you have a `test.txt-e` file now though?

---

Oh, wait, I see it now. There also needs to be a space between `-i ''`. Cool, this saves me from using `find . -name *.bak | xargs rm` on MacOS in the future.

---

No, there is no `test.txt-e` file created. That happens because `-i` in BSD sed expects an argument (unlike GNU sed). So if you do 

```
sed -i -e 's/abc/def/g' test.txt
```

`-i` gobbles `-e` and uses it as an extension. So you need an empty place holder `''` to suppress this.